docs/xclim_migration_guide.rst

```diff
@@ -0,0 +1,52 @@
+=====================
+xclim Migration Guide
+=====================
+
+`xsdba` was first developed as a submodule of `xclim`_. The reason of the split was twofold
+
+* `xsdba` grew as a heavy submodule within `xclim` with a different aim;
+* To increase collaboration with other consortiums, we want more flexibility (i.e. this will break more often)
+
+For `xclim` users, it will still be possible to import `sdba` if `xsdba` is installed.
+
+.. code-block:: python
+
+    from xclim import sdba
+
+
+Units handling
+--------------
+
+One important change concerns units handling.  Conversion between units with different dimensionality was previously automated, when possible, in `xclim.sdba`, given that the target dataset had a suitable standard name (e.g. allowing conversion of `lwe_precipitation_rate` from `kg m-2 s-1` to `mm d-1`). `xsdba` will not perform these automatic conversions, thus the datasets and thresholds to be compared should be given in units that share the same dimensionality before being passed to `xsdba` functions
+
+ `xsdba` still implements some unit handling through `pint`. High level adjustment objects will (usually) parse units from the passed xarray objects, check that the different inputs have matching units and perform conversions when units are compatible but don't match. `xsdba` imports `cf-xarray's unit registry <https://cf-xarray.readthedocs.io/en/latest/units.html>`_ by default and, as such, expects units matching the `CF conventions <https://cfconventions.org/Data/cf-conventions/cf-conventions-1.12/cf-conventions.html#units>`_. This is very similar to `xclim`, except for two features:
+
+    - `xclim` is able to detect more complex conversions based on the CF "standard name". `xsdba` will not implement this, the user is expected to provide coherent inputs to the adjustment objects.
+    - `xclim` has a convenience "hydro" context that allows for conversion between rates and fluxes, thicknesses and amounts of liquid water quantities by using an implicit the water density (1000 kg/m³).
```

Comment:
The first paragraph kind of repeats some information found in the list, but I think it made it clear to start with a paragraph that summarizes what changes and how to use xsdba from now on. If that migration guide looks all right we'll be good to go

Reply:
Added a few small changes. Assuming everything builds, I think we're good.